### ISSUE TITLE
Context.js onload() not called with browser refresh or source editor changes

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#2808]
Assigning to @peller because he is the expert with the complex web of deferred's that occur when a page is opened, but others (including me) might be able to figure this one out.

Part of document initialization involves calling the "onload" function in Context.js. If you open a file from Files palette, then the onload() function gets called. But if you refresh your browser, the open editor initialization process never calls "onload", and if you make changes in the source editor, the "onload" function isn't called, despite the fact that _processWidgets() gets called, and its "then" clause gets invoked, which says:

```
            connect.connect(this.getGlobal(), 'onload', this, function() {
                this.onload();
            });
```

It looks like there is a timing issue with the `onload` event where the event is firing before the then clause from processWidgets() gets invoked.

**Comment by peller** [comment]
I'm thinking this should be replaced with a call to dojo/ready.  onload used to work in a sync world where everything blocked on one thread.

**Comment by peller** [comment]
actually, it was f51f2f59b07aa104a0e5bc209c845f53629d2099 which changed the onload logic, moving the connect until after widget processing.  I think what was intended was to defer firing the event until both widget processing and page load were completed, rather than just deferring the registration.

**Comment by billreed63** [comment]
This issue has returned, onload is not gettng called at all on html page load which is the root cause of issue #3031

**Comment by peller** [comment]
could this have been fixed by d323e84d35da0eb1a1ca245ee0b16e9d02551203?

**Comment by JonFerraiolo** [comment]
I put a breakpoint at Context.js:1298 (the `connect.connect` line below) and Context.js:1299 (the `this.onload()` line below):

```
        return this._processWidgets(containerNode, active, this._loadFileStatesCache, scripts).then(function(){
            connect.connect(this.getGlobal(), 'onload', this, function() {
                this.onload();
            });
        }.bind(this));
```

and I'm hitting the breakpoint at `connect.connect` but not `this.onload`. My guess is that because of our deferred logic, the `connect.connect` event is registering an onload handler after the load event has already happened.
